### PR TITLE
docs[patch]: update chatbot tutorial and migration guide

### PR DESCRIPTION
**Pull request by ccurme**

### CODE REVIEW

File: docs/docs/tutorials/chatbot.ipynb [hunk 1]
```diff
@@ -270,29 +255,33 @@
...
-    "        store[session_id] = InMemoryChatMessageHistory()\n",
-    "    return store[session_id]\n",
+    "# Define the function that calls the model\n",
+    "def call_model(state: MessagesState):\n",
+    "    response = model.invoke(state[\"messages\"])\n",
+    "    return {\"messages\": response}\n",
     "\n",
     "\n",
-    "with_message_history = RunnableWithMessageHistory(model, get_session_history)"
+    "# Define the two nodes we will cycle between\n",
```
Comment (vbarda) [inline docs/docs/tutorials/chatbot.ipynb:271]: this is a typo

File: docs/docs/tutorials/chatbot.ipynb [hunk 2]
```diff
@@ -335,30 +334,31 @@
...
    "source": [
-    "response = with_message_history.invoke(\n",
-    "    [HumanMessage(content=\"What's my name?\")],\n",
-    "    config=config,\n",
-    ")\n",
+    "query = \"What's my name?\"\n",
     "\n",
-    "response.content"
+    "input_messages = [HumanMessage(query)]\n",
+    "for event in app.stream({\"messages\": input_messages}, config, stream_mode=\"values\"):\n",
```
Comment (vbarda) [inline docs/docs/tutorials/chatbot.ipynb:353]: should we keep .invoke here?

File: docs/docs/tutorials/chatbot.ipynb [hunk 3]
```diff
@@ -35,6 +35,8 @@
     "\n",
     ":::\n",
     "\n",
+    "*Note: this tutorial previously built a chatbot using [RunnableWithMessageHistory](https://python.langchain.com/api_reference/core/runnables/langchain_core.runnables.history.RunnableWithMessageHistory.html). You can access this version of the tutorial in the [v0.2 docs](https://python.langchain.com/v0.2/docs/tutorials/chatbot/).*\n",
```
Comment (vbarda) [inline docs/docs/tutorials/chatbot.ipynb:38]: should this be an admonition?

File: docs/docs/tutorials/chatbot.ipynb (hunk 1 above)
Comment (ccurme) [reply docs/docs/tutorials/chatbot.ipynb]: Thank you!

File: docs/docs/tutorials/chatbot.ipynb (hunk 2 above)
Comment (ccurme) [reply docs/docs/tutorials/chatbot.ipynb]: Updated everywhere

File: docs/docs/tutorials/chatbot.ipynb (hunk 3 above)
Comment (ccurme) [reply docs/docs/tutorials/chatbot.ipynb]: Updated

**Pull request merged**